src/villagebook/organisation/handlers.clj

```diff
@@ -0,0 +1,25 @@
+(ns villagebook.organisation.handlers
+  (:require [ring.util.response :as res]
+
+            [villagebook.organisation.db :as db]
+            [villagebook.organisation.spec :as organisation-spec]
+            [clojure.spec.alpha :as s]))
+
+(defn create-organisation
+  [request]
+  (let [{name :name
+         color :color
+         :as orgdata} (:params request)]
+
+    (if (organisation-spec/valid-organisation-details? orgdata)
+      (let [neworg (db/create-organisation orgdata)]
```

Comment:
DB exceptions should be caught. Consider handling with

```clojure
{:error e}
```

Reply:
@neenaoffline Should I add the try-catch to the db layer? I'd have to check for the ```{:error e}``` at all layers above this and then generate a 500 at the handler, instead of the current setup where ring automatically generates the 500.

Reply:
@prabhanshuguptagit Maybe let's leave it as it is.